sos/plugins/cnv.py

```diff
@@ -0,0 +1,234 @@
+                for s in subcmds_dep:
+                    self.add_cmd_output(["%s " % oc_cmd + s + " " + d])
+        for svc in all_svc:
+            for sv in self.get_resc_ns(self.get_cmd_output_now(svc)):
+                for s in subcmds_svc:
+                    self.add_cmd_output(["%s " % oc_cmd + s + " " + sv])
+        for pod in all_pods:
+            for p in self.get_resc_ns(self.get_cmd_output_now(pod)):
+                for s in subcmds_pods:
+                    self.add_cmd_output(["%s " % oc_cmd + s + " " + p])
+        for stor in all_stor:
+            for pvc in self.get_resc_ns(self.get_cmd_output_now(stor)):
+                self.add_cmd_output(["%s get pvc -o yaml " % oc_cmd + pvc])
+        self.add_cmd_output("%s get vm --all-namespaces" % oc_cmd)
+        vmis = self.get_cmd_output_now("%s get vmi --all-namespaces" % oc_cmd)
+
+        if self.get_option("vm"):
+            options = self.get_option("vm")
+            if isinstance(options, bool):
+                return
+
+            subcmds_vm = [
+                "describe vm",
+                "describe vmi",
+                "get vm -o yaml",
+                "get vmi -o yaml",
+            ]
+            node_pods = [
+                "%s get pods -l kubevirt.io=virt-handler "
+                "--all-namespaces -o wide" % oc_cmd,
+                "%s get pods -l app=multus "
+                "--all-namespaces -o wide" % oc_cmd,
+                "%s get pods -l app=ovs-cni "
+                "--all-namespaces -o wide" % oc_cmd,
+                "%s get pods -l app=ovs "
+                "--all-namespaces -o wide" % oc_cmd,
+                "%s get pods -l app=sdn "
+                "--all-namespaces -o wide" % oc_cmd,
+            ]
+
+            for vm in self.get_vm_ns(vmis, self.get_option("vm")):
+                for s in subcmds_vm:
+                    self.add_cmd_output(["%s " % oc_cmd + s + " " + vm])
+                getvmi = "%s get vmi " % oc_cmd + vm + " --show-labels"
+                lbl_ext = self.call_ext_prog(getvmi)
+                vmi_l = ""
+                if lbl_ext['status'] != 0:
+                    return
+                for line in lbl_ext['output'].splitlines():
+                    if line.split()[0] == vm.split()[0]:
+                        for l in line.split()[2].split(','):
+                            if vm.split()[0] in l:
+                                vmi_l = "-l " + l + " --all-namespaces -o wide"
+                p_ln = self.get_cmd_output_now("%s get pods " % oc_cmd + vmi_l)
+                for pod in self.get_resc_ns(p_ln):
+                    for s in subcmds_pods:
+                        self.add_cmd_output(["%s " % oc_cmd + s + " " + pod])
+                    exc = "%s " % oc_cmd + "exec " + pod + " -- virsh dumpxml "
+                    dmn = vm.split()[2] + "_" + vm.split()[0]
+                    self.add_cmd_output([exc + dmn])
+                node = self.get_node_info(p_ln)
```

Comment:
This all seems a bit heavy for vm pod names, especially considering the kubernetes plugin already handles the collection of items that we're getting the pod names for later on around line 215.

Surely there's a more direct way of identifying vm pods?